packages/atomic/src/components/commerce/product-template-components/atomic-product-children/atomic-product-children.tsx

```diff
@@ -0,0 +1,188 @@
+  // TODO: remove this prop; we'll want the carousel to automatically kick in at the proper responsive breakpoint instead.
+  @Prop() public childrenPerPage: number = 6;
+
+  /**
+   * The non-localized label to display for the product children section.
+   */
+  @Prop() public label: string = 'available-in';
+
+  @Event({
+    eventName: 'atomic/hoverChildProduct',
+    bubbles: true,
+  })
+  public hoverChildProduct!: EventEmitter<{
+    hoveredChildProduct: ChildProduct;
+    currentParentProductId: string;
+    originalParentProduct: Product;
+  }>;
+
+  constructor() {
+    this.validateProps();
+  }
+
+  public connectedCallback(): void {
+    this.originalParentProduct = this.isProduct(this.product.children[0])
+      ? this.product.children[0]
+      : this.product;
+    this.children = [
+      this.originalParentProduct,
+      ...this.originalParentProduct.children,
+    ];
+  }
+
+  private isProduct(child: Product | ChildProduct): child is Product {
+    return 'children' in child;
+  }
+
+  private validateProps() {
+    new Schema({
+      label: new StringValue(),
+    }).validate({
+      label: this.label,
+    });
+  }
+
+  private onHoverChild(
+    hoveredChildProduct: ChildProduct,
+    currentParentProductId: string
+  ) {
+    this.hoverChildProduct.emit({
+      hoveredChildProduct,
+      currentParentProductId,
+      originalParentProduct: this.originalParentProduct ?? this.product,
+    });
+  }
+
+  /**
+   * Moves to the previous page, when the carousel is activated.
+   */
+  @Method() public async previousPage() {
+    this.currentPage =
+      this.currentPage - 1 < 0 ? this.numberOfPages - 1 : this.currentPage - 1;
+  }
+
+  /**
+   * Moves to the next page, when the carousel is activated.
+   */
+  @Method() public async nextPage() {
+    this.currentPage = (this.currentPage + 1) % this.numberOfPages;
+  }
+
+  private get numberOfPages() {
+    return Math.ceil(this.children.length / this.childrenPerPage!);
+  }
+
+  private get currentIndex() {
+    return Math.abs(
+      (this.currentPage * this.childrenPerPage) % this.children.length
+    );
+  }
+
+  private get subsetOfChildren() {
+    if (!this.childrenPerPage) {
+      return this.children;
+    }
+
+    return this.children.slice(
+      this.currentIndex,
+      this.currentIndex + this.childrenPerPage
+    );
+  }
+
+  private renderCarousel() {
+    return (
+      <Carousel
+        bindings={this.bindings}
+        currentPage={this.currentPage}
+        nextPage={() => this.nextPage()}
+        previousPage={() => this.previousPage()}
+        numberOfPages={this.numberOfPages}
+      >
+        {this.subsetOfChildren.map((child) => this.renderChild(child))}
+      </Carousel>
+    );
+  }
+
+  // TODO: get rid of most of the inline styles; in most cases we'll use the generic stuff.
```

Comment:
I would do it right now, or wait for product sections to guide you before merging this then.
As discussed in person, template components need to take all available space given to them, not limit their own size/width/height themselves (as a general rule of thumb)

It's the responsability of `product-sections` to define the space available.